pandas/io/formats/style.py

```diff
@@ -1294,6 +1307,40 @@ def hide_columns(self, subset) -> Styler:
         self.hidden_columns = self.columns.get_indexer_for(hidden_df.columns)
         return self
 
+    def _default_formatter(self, x):
+        if isinstance(x, (float, complex)):
+            return f"{x:.{self.precision}f}"
+        return x
+
+    def _maybe_wrap_formatter(
```

Comment:
why is this moved? what is the point of self.na_rep?

Reply:
its moved because as a module level it can't act dynamically and update `self.na_rep` if it is later changed. At class level it can so this does to inheritance.